workshops/lyon2018/intro-with-globular.ipynb

```diff
@@ -928,16 +971,105 @@
    "outputs": [],
    "source": [
     "display2.mtv(residuals, title=\"residuals\")\n",
+    "display2.setMaskTransparency(5)\n",
+    "display2.setMaskTransparency(30, 'DETECTED')\n",
     "display3.mtv(model, title=\"model\")"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "display2.scale('linear', -0.05, 2)"
+   ]
+  },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "Even better!  One more iteration *might* make things better, but we're probably getting to the limits of this very simple algorithm (especially since there really are a lot of galaxies in this image)."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Uploading the catalog directly to Firefly"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As noted earlier, Firefly has sophisticated capabilities for viewing tables and overlaying catalogs. The `firefly_client` Python package underlies the Firefly backend in `lsst.afw.display`. The `firefly_client.plot` module includes convenience functions that we'll demonstrate here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import firefly_client.plot as ffplt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ensure we use the same setup as our Display instances:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ffplt.use_client(display._client)"
```

Comment:
If one already gets a display earlier, does s/he still need to run this line?

Reply:
Excellent question. The intent in `firefly_client.plot` is to make this line unnecessary when an `afw.display` has already been defined.  We are tracking instances of `FireflyClient` inside the class of the same name. When I had tried using `firefly_client.plot` last week on lsp-demo, it came up with a different channel name which would need a different display.

Here is the code in the `plot` module. I will need to figure out why this didn't work for me on lsp-demo:
```
fc = None

for c in FireflyClient.get_instances():
    _ = c()
    if _ and _._is_page_connected():
        fc = _
        break
```

Reply:
Upon re-testing: it works without this line! Maybe last week I had shut down my browser tab and not had the IFrame connected, which would have caused the `_is_page_connected()` to return False.

I'll note this in the Markdown cell and I'll comment out this line.